src/main/java/com/epam/tc/hw6/grid/DriverFabric.java

```diff
@@ -0,0 +1,57 @@
+package com.epam.tc.hw6.grid;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+import io.github.bonigarcia.wdm.config.WebDriverManagerException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Locale;
+import lombok.NonNull;
+import lombok.experimental.UtilityClass;
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.MutableCapabilities;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.firefox.FirefoxOptions;
+import org.openqa.selenium.opera.OperaOptions;
+import org.openqa.selenium.remote.BrowserType;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+
+@UtilityClass
+public class DriverFabric {
+
+    public WebDriver getWebDriver(final boolean isLocal, final String hub, @NonNull final String browser) {
+
+        if (isLocal) {
+            return WebDriverManager.getInstance(browser).create();
+        } else {
+            Capabilities caps = getCapabilities(browser);
+            URL hubAddress = getUrl(hub);
+            return new RemoteWebDriver(hubAddress, caps);
+        }
+    }
+
+    private URL getUrl(String hub) {
+        try {
+            return new URL(hub);
+        } catch (MalformedURLException e) {
+            throw new RuntimeException("Hub URL looks malformed", e);
+        }
+    }
+
+    private MutableCapabilities getCapabilities(String browserName) {
+        switch (browserName.toLowerCase(Locale.ROOT)) {
+            case BrowserType.FIREFOX:
+                return new FirefoxOptions();
```

Comment:
I have some local problem. This is a fix:
```
            case BrowserType.FIREFOX:
                FirefoxOptions firefoxOptions = new FirefoxOptions();
                firefoxOptions.setBinary("C:\\Program Files\\Mozilla Firefox\\firefox.exe");
                return firefoxOptions;
```
It's just for somebody who try to run your test. Your code is great! No need to change anything!